sys-devel/llvm-roc/llvm-roc-5.0.2.ebuild

```diff
@@ -0,0 +1,95 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake
+
+DESCRIPTION="Radeon Open Compute llvm,lld,clang"
+HOMEPAGE="https://github.com/RadeonOpenCompute/ROCm/"
+SRC_URI="https://github.com/RadeonOpenCompute/llvm-project/archive/rocm-${PV}.tar.gz -> llvm-rocm-ocl-${PV}.tar.gz"
+
+LICENSE="UoI-NCSA rc BSD public-domain"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="debug +runtime"
+
+RDEPEND="
+	dev-libs/libxml2
+	sys-libs/zlib
+	sys-libs/ncurses:="
+DEPEND="${RDEPEND}"
+
+S="${WORKDIR}/llvm-project-rocm-${PV}/llvm"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-4.2.0-current_pos.patch"
+	"${FILESDIR}/${PN}-5.0.0-linkdl.patch"
+)
+
+CMAKE_BUILD_TYPE=RelWithDebInfo
+
+src_prepare() {
+	cd "${WORKDIR}/llvm-project-rocm-${PV}" || die
+	# eapply "${FILESDIR}/${PN}-3.0.0-add_libraries.patch"
+	eapply "${FILESDIR}/${PN}-4.0.0-remove-isystem-usr-include.patch"
+	eapply "${FILESDIR}/${PN}-5.0.0-hip-location.patch"
+	eapply "${FILESDIR}/${PN}-5.0.0-add_GNU-stack.patch"
+
+	if [[ -n ${EPREFIX} ]]; then
+		pushd "${S}"/../clang >/dev/null || die
+		sed -i -e "s@DEFAULT_SYSROOT \"\"@DEFAULT_SYSROOT \"${EPREFIX}\"@" CMakeLists.txt || die
+		eend $?
+		ebegin "Use "${EPREFIX}" as default sysroot"
+		cd lib/Driver/ToolChains >/dev/null || die
+		ebegin "Use dynamic linker from ${EPREFIX}"
+		sed -i -e "/LibDir.*Loader/s@return \"\/\"@return \"${EPREFIX}/\"@" Linux.cpp || die
+		eend $?
+
+		ebegin "Remove --sysroot call on ld for native toolchain"
+		sed -i -e "$(grep -n -B1 sysroot= Gnu.cpp | sed -ne '{1s/-.*//;1p}'),+1 d" Gnu.cpp || die
+		eend $?
+		popd >/dev/null || die
+	fi
+
+	# handled by sysroot, don't prefixify here.
+	sed -e 's:/opt/rocm:/usr/lib/hip:' \
+		-i "${S}"/../clang/lib/Driver/ToolChains/AMDGPU.cpp || die
+
+	eapply_user
+	cmake_src_prepare
+}
+
+src_configure() {
+	PROJECTS="clang;lld"
+
+	if usex runtime; then
+		PROJECTS+=";compiler-rt"
+	fi
+
+	local mycmakeargs=(
+		-DCMAKE_INSTALL_PREFIX="${EPREFIX}/usr/lib/llvm/roc"
+		-DCMAKE_DISABLE_FIND_PACKAGE_CUDA=ON
+		-DLLVM_ENABLE_PROJECTS="${PROJECTS}"
+		-DLLVM_TARGETS_TO_BUILD="AMDGPU;X86"
+		-DLLVM_BUILD_DOCS=NO
+		-DLLVM_ENABLE_OCAMLDOC=OFF
+		-DLLVM_ENABLE_SPHINX=NO
+		-DLLVM_ENABLE_DOXYGEN=OFF
+		-DLLVM_INSTALL_UTILS=ON
+		-DLLVM_VERSION_SUFFIX=roc
+		-DOCAMLFIND=NO
+	)
+
+	use debug || local -x CPPFLAGS="${CPPFLAGS} -DNDEBUG"
+
+	cmake_src_configure
+}
```

Comment:
Running a diff against llvm-4.5.2 resulted
```
 src_configure() {                                                                                        
-       PROJECTS="clang;lld;llvm"                                                                         
+       PROJECTS="clang;lld"                         
                                                                                                          
        if usex runtime; then                        
                PROJECTS+=";compiler-rt"             
        fi                                                                                                
                                                                                                          
        local mycmakeargs=(                                                                               
-               -Wno-dev                                                                                  
                -DCMAKE_INSTALL_PREFIX="${EPREFIX}/usr/lib/llvm/roc"                                      
                -DCMAKE_DISABLE_FIND_PACKAGE_CUDA=ON                                                      
                -DLLVM_ENABLE_PROJECTS="${PROJECTS}"                                                      
                -DLLVM_TARGETS_TO_BUILD="AMDGPU;X86"                                                      
                -DLLVM_BUILD_DOCS=NO                                                                      
-               -DLLVM_ENABLE_BINDINGS=OFF                                                                
                -DLLVM_ENABLE_OCAMLDOC=OFF           
                -DLLVM_ENABLE_SPHINX=NO              
                -DLLVM_ENABLE_DOXYGEN=OFF                                                                 
-               -DLLVM_BUILD_UTILS=ON                                                                     
+               -DLLVM_INSTALL_UTILS=ON                                                                   
                -DLLVM_VERSION_SUFFIX=roc                                                           
                -DOCAMLFIND=NO                       
        )                                 
```
Could you please explain them in the commit message?

Reply:
> Running a diff against llvm-4.5.2 resulted
> 
> ```
>  src_configure() {                                                                                        
> -       PROJECTS="clang;lld;llvm"                                                                         
> +       PROJECTS="clang;lld"                         
>                                                                                                           
>         if usex runtime; then                        
>                 PROJECTS+=";compiler-rt"             
>         fi                                                                                                
>                                                                                                           
>         local mycmakeargs=(                                                                               
> -               -Wno-dev                                                                                  
>                 -DCMAKE_INSTALL_PREFIX="${EPREFIX}/usr/lib/llvm/roc"                                      
>                 -DCMAKE_DISABLE_FIND_PACKAGE_CUDA=ON                                                      
>                 -DLLVM_ENABLE_PROJECTS="${PROJECTS}"                                                      
>                 -DLLVM_TARGETS_TO_BUILD="AMDGPU;X86"                                                      
>                 -DLLVM_BUILD_DOCS=NO                                                                      
> -               -DLLVM_ENABLE_BINDINGS=OFF                                                                
>                 -DLLVM_ENABLE_OCAMLDOC=OFF           
>                 -DLLVM_ENABLE_SPHINX=NO              
>                 -DLLVM_ENABLE_DOXYGEN=OFF                                                                 
> -               -DLLVM_BUILD_UTILS=ON                                                                     
> +               -DLLVM_INSTALL_UTILS=ON                                                                   
>                 -DLLVM_VERSION_SUFFIX=roc                                                           
>                 -DOCAMLFIND=NO                       
>         )                                 
> ```

@inode64 Could you explain those changes when bumping llvm-roc from 4.3.0-r1 to 4.5.2? I think both of the configurations are OK.